src/Client.php

```diff
@@ -205,19 +206,33 @@ public function flush(?int $timeout = null): PromiseInterface
     /**
      * Assembles an event and prepares it to be sent of to Sentry.
      *
-     * @param array<string, mixed>|Event $payload The payload that will be converted to an Event
-     * @param Scope|null                 $scope   Optional scope which enriches the Event
+     * @param Event          $event The payload that will be converted to an Event
+     * @param EventHint|null $hint  May contain additional information about the original exception
+     * @param Scope|null     $scope Optional scope which enriches the Event
      *
      * @return Event|null The prepared event object or null if it must be discarded
      */
-    private function prepareEvent($payload, ?Scope $scope = null): ?Event
+    private function prepareEvent(Event $event, ?EventHint $hint = null, ?Scope $scope = null): ?Event
     {
-        if ($this->options->shouldAttachStacktrace() && !($payload instanceof Event) && !isset($payload['exception']) && !isset($payload['stacktrace'])) {
-            $event = $this->buildEventWithStacktrace($payload);
-        } else {
-            $event = $this->buildEvent($payload);
+        if (null !== $hint) {
+            if (null !== $hint->exception && empty($event->getExceptions())) {
+                $this->addThrowableToEvent($event, $hint->exception);
+            }
+
+            if (null !== $hint->stacktrace && null === $event->getStacktrace()) {
```

Comment:
Shouldn't the stacktrace be set only when no exception is set? I remember I read something like this somewhere in the Unified API docs, not sure where thought

Reply:
You are correct at least according to the old code.

Reply:
This doesn't looks resolved to me. You changed the code in the `addMissingStacktraceToEvent` method to handle this, but here you are still setting the stacktrace regardless